### ISSUE TITLE
Heightmap Bug after walking a bigger distance

### ISSUE THREAD

**Issue by Narrenschlag** [mapbox/mapbox-unity-sdk#1687]
My problem is that after the player travels a bigger distance away from where he started the application the objects in worldspace start to spawn in the air. 

* Unity version: 2019.4.10
* Mapbox SDK version: 2.1.1
* The platform you're building to: Android
* A description of what you're trying to do: Location Based Game
* Steps to recreate the bug if appropriate: **_>Activate Elevated Terrain>>>Travel a far distance_**

So let's see...

![image](https://user-images.githubusercontent.com/71547522/95427815-7038d480-0948-11eb-9fb8-be2fc2447ba4.png)

My problem is that after the player travels a bigger distance away from where he started the application the objects in worldspace start to spawn in the air. 
You can restart the application to fix it so it doesn't depend on local heightmap issues. :/
This happens with my trees and some rare times with my streets and buildings aswell.
I found out that when you travel in the editor with a high speed you can see how the elevated terrain is first generated with a far higher ground level than normal.

Does anybody know what's wrong?

P.S.: What is the "Disposal Buffer" for?

Greetings
Max

**Comment by Narrenschlag** [comment]
@brnkhy u got an idea for a quick fix?

Here a gif of some other appearances of the bug I mean. I think that the elevation is the problem. Somebody got a fix? Cause I really wanna keep elevated terrain in the game :(

![ragnarok_mapbug1](https://user-images.githubusercontent.com/71547522/95431484-8301d800-094d-11eb-9cce-ffd15204554a.gif)

**Comment by Narrenschlag** [comment]
Also it seems like there are multiple tiles loaded at once at the same position. but only one of them belongs there.
Maybe some problem with the recycling? As I already said this normally happens only when playing for a while.

Her another example

![image](https://user-images.githubusercontent.com/71547522/95441731-c9a9ff00-095a-11eb-9558-c03c519f722a.png)

**Comment by Narrenschlag** [comment]
It really seems to be from a different tile, because it use a completly different terrain to place the trees.
![image](https://user-images.githubusercontent.com/71547522/95445326-31624900-095f-11eb-99a0-f7322f30e0dc.png)

**Comment by Narrenschlag** [comment]
### Summary: 
**There is some problem with the "Map Layers >>> Features". The terrain itself spawns correctly.**
The problem comes in when the player moves away from the start position of the map. Objects like buildings, streets and points spawn inside the new tiles with the local position they had in their original tile. 

### The Question:
**Is there a way to prevent this "recycling-like" behaviour? Because I don't want this to happen.**

**Comment by Narrenschlag** [comment]
Update:
I increased the disposal buffer from 5 > 100. That seems to help a bit with the label problems.

**Comment by brnkhy** [comment]
Hey @Narrenschlag,
it might be an issue with recycling indeed. Looks like everything snaps well but to a different elevation/layer. But actually with trees, it might also be that old trees aren't destroyed.
Can you check if those buildings floating in air are supposed to be there in that tile? I mean does those buildings really exists and snapped to wrong elevation or are they just leftovers from another tile? Because I think I see bunch of buildings snap perfectly well and only a bunch in the air.

**Comment by Narrenschlag** [comment]
@brnkhy No, the floating objects are not supposed to be in that specific tile. They are from different already loaded tiles. Some of the objects you can see on the map are supposed to but not all of them are.

Greetings

Max

**Comment by Narrenschlag** [comment]
@brnkhy, 
no offense and I don't want to agitate you, 
but what can I do to fix this issue?
Walking big distances with the app active is a key aspect of the game and I need it working as soon as possible.

Greetings

Max

**Comment by brnkhy** [comment]
Hey @Narrenschlag!
I'm sorry I'm overwhelmed these days and I thought I actually posted this to you as well (I did to another ticket, confused this with that). I know it's horrible to post it as an image but it's just 2-3 lines and it was the fastest way. Can you please let me know if it fixes the issue?
![msedge_5cSp9OESpS](https://user-images.githubusercontent.com/1465390/99987277-53a21000-2dc1-11eb-8776-9d94fe6f477e.png)

**Comment by brnkhy** [comment]
I just noticed this change is already in dev the dev branch of SDK but I believe it wasn't in the release package. 
https://github.com/mapbox/mapbox-unity-sdk/commit/b9a5acd69e74b966c10512a4f37b4f6aecade4d6#diff-1d49886de80eb34d12e1a3b63444abb41ae7fd1dda7d9134b399125965848b88

**Comment by Narrenschlag** [comment]
Nope it is still not working
But thank you 

Greetings

Max

**Comment by Narrenschlag** [comment]
After some time some objects won't even spawn anymore and old objects like trees and points stay. :/

**Comment by Narrenschlag** [comment]
@brnkhy , It seems that this fixes only the buildings, streets and water. Because they get their correct position.

But things like trees that are "SpawnInside" objects or points of poi_labels get messed up.

Got a good feeling that we almost got it! 

Greetings Max 😃
